builder/ibmcloud/vpc/step_get_subnet_info.go

```diff
@@ -43,6 +45,63 @@ func (s *stepGetSubnetInfo) Run(_ context.Context, state multistep.StateBag) mul
 	ui.Say(fmt.Sprintf("VPC ID: %s", vpcId))
 	ui.Say(fmt.Sprintf("Zone: %s", zone))
 
+	if config.CatalogOfferingCRN != "" || config.CatalogOfferingVersionCRN != "" || config.EncryptionKeyCRN != "" {
+		// validate crn
+
+		searchURL := "https://api.global-search-tagging.cloud.ibm.com"
+		globalSearchV2Options := &searchv2.GlobalSearchV2Options{
+			URL:           searchURL,
+			Authenticator: vpcService.Service.Options.Authenticator,
+		}
+		globalSearchAPIV2, err := searchv2.NewGlobalSearchV2(globalSearchV2Options)
+
+		if err != nil {
+			fmt.Println("Gen2 Service creation failed.", err)
+		}
+		// validate catalog offering crn
+		if config.CatalogOfferingCRN != "" {
+			crnToCheck := fmt.Sprintf("%s%s", strings.Split(config.CatalogOfferingCRN, ":offering")[0], "::")
+			query := fmt.Sprintf("crn:\"%s\"", crnToCheck)
+			isHidden := "any"
+			searchOptions := &searchv2.SearchOptions{
+				Query:    &query,
+				IsHidden: &isHidden,
+			}
+			res, _, _ := globalSearchAPIV2.Search(searchOptions)
+			if len(res.Items) != 0 {
+				ui.Say(fmt.Sprintf("%s Catalog information successfully retrieved ...", res.Items[0].GetProperty("name")))
+			}
+		}
+		// validate catalog version crn
+		if config.CatalogOfferingVersionCRN != "" {
+			crnToCheck := fmt.Sprintf("%s%s", strings.Split(config.CatalogOfferingVersionCRN, ":version")[0], "::")
+			query := fmt.Sprintf("crn:\"%s\"", crnToCheck)
+			isHidden := "any"
+			searchOptions := &searchv2.SearchOptions{
+				Query:    &query,
+				IsHidden: &isHidden,
+			}
+			res, _, _ := globalSearchAPIV2.Search(searchOptions)
+			if len(res.Items) != 0 {
+				ui.Say(fmt.Sprintf("%s Catalog information successfully retrieved ...", res.Items[0].GetProperty("name")))
+			}
+		}
+		// validate encryption key crn
+		if config.EncryptionKeyCRN != "" {
+			crnToCheck := fmt.Sprintf("%s%s", strings.Split(config.EncryptionKeyCRN, ":key")[0], "::")
+			query := fmt.Sprintf("crn:\"%s\"", crnToCheck)
+			isHidden := "any"
+			searchOptions := &searchv2.SearchOptions{
+				Query:    &query,
+				IsHidden: &isHidden,
+			}
+			res, _, _ := globalSearchAPIV2.Search(searchOptions)
+			if len(res.Items) != 0 {
+				ui.Say(fmt.Sprintf("%s Encryption information successfully retrieved ...", res.Items[0].GetProperty("name")))
+			}
```

Comment:
What if global catalog does not contain the CRN? Need to put in error handling for if no item is found